content/blogs/2024-02-28-dbt-or-sqlmesh.md

```diff
@@ -0,0 +1,103 @@
+
+Talking about dependencies: we often have more complex pipelines than what we see in Modern Data Stack schemas where it’s often about connecting an ingesting tool ([Airbyte](https://airbyte.com/), [Fivetran](https://www.fivetran.com), [dlt](https://dlthub.com/)), transforming tool (dbt), and a reverse ETL tool ([Hightouch](https://hightouch.com/), [Census](https://www.getcensus.com/reverse-etl)) or a dashboarding tool ([Tableau](https://www.tableau.com), [PowerBI](https://www.microsoft.com/en-en/power-platform/products/power-bi), [Metabase](https://www.metabase.com/), [Superset](https://superset.apache.org/)). Data teams also need notification services, custom API calls, monitoring, very specific transformations, etc.
+
+Connecting all those tools needs a control plane - [an orchestrator](https://github.com/kestra-io/kestra).
+
+## Is it time to say goodbye to dbt and move to SQLMesh?
+
+At the end of 2022, ex-engineers from Airbnb, Apple, Google, and Netflix started a project called [SQLMesh](https://sqlmesh.com/). Built on top the the SQLGlot library allowing parsing and transpiling SQL SQLMesh is, like dbt, a framework to run data pipelines written in SQL. The major shift from dbt is that SQLMesh has an emphasis on operations.
+
+Major differences with dbt are:
+- SQLMesh can infer query dependencies
+- It can create new environments without duplicating data. Allowing to create dynamic representations of the data while ensuring tables are never built more than once. Unit tests, Audits, and Data Diff provide validation throughout the development workflow.
+- It comes with different user interfaces: a web UI and a CLI. Both are in the open-source version.
+
+You can find all the differences explained by the SQLMesh team [in this documentation](https://sqlmesh.readthedocs.io/en/stable/comparisons/#feature-comparisons).
+
+![sqlmesh-ui](/blogs/2024-02-27-dbt-or-sqlmesh/sqlmesh-ui.png)
+
+![sqlmesh-kestra](/blogs/2024-02-27-dbt-or-sqlmesh/sqlmesh-kestra.png)
+
+## Alternatives
+
+Transforming data with SQL is in growing need, and several other actors have to be considered in that space:
+
+- [Y42](https://www.y42.com/blog/virtual-data-builds-one-data-warehouse-environment-for-every-git-commit), [GCP Dataform](https://cloud.google.com/dataform), or [SDF](https://www.sdf.com/):  they are dbt Cloud alternatives (no open-source versions) with their advantages and limitations.
+
+![y42](/blogs/2024-02-27-dbt-or-sqlmesh/y42.png)
+
+- [lea](https://github.com/carbonfact/lea): a lightweight alternative to dbt or SQLMesh. Fully open-source and maintained by the team using it for their data warehouse ([CarbonFact](https://www.carbonfact.com/)).
+
+- [Quary](https://www.quary.dev/): an open-source alternative of dbt Core, made in Rust.
+
+- Building your own “in-house dbt” is definitely something to consider if you have the skill and the need for a custom solution. dbt is basically SQL plus templating, so why not build your own?
```

Comment:
```suggestion
```

please don't even suggest it 😅 bad idea to reinvent the wheel and build a worse version of what already exists